db/migrate/20210126112238_add_uniqueness_constrain_on_lastname_for_groups_and_placeholder_users.rb

```diff
@@ -0,0 +1,17 @@
+class AddUniquenessConstrainOnLastnameForGroupsAndPlaceholderUsers < ActiveRecord::Migration[6.1]
+  def up
+    # Partial unique index
+    execute <<-SQL
+      CREATE UNIQUE INDEX unique_lastname_for_groups_and_placeholder_users ON
+        users (lastname, type)
+        WHERE (type = 'Group' OR type = 'PlaceholderUser');
+    SQL
```

Comment:
This is guesswork as I did not really check the scenarios in which it might be used but I'd guess that it is more beneficial to switch the order of the fields to `type, lastname` as I'd imagine first checking by type will reduce the records faster.

Reply:
Isn't it the other way round? `lastname` is much more specific than the `type` as there are much more lastnames/groupnames/placeholdernames than types.